src/main/java/baritone/behavior/LookBehavior.java

```diff
@@ -20,111 +20,209 @@
                 break;
         }
     }
 
+    @Override
+    public void onSendPacket(PacketEvent event) {
+        if (!(event.getPacket() instanceof CPacketPlayer)) {
+            return;
+        }
+
+        final CPacketPlayer packet = (CPacketPlayer) event.getPacket();
+        if (packet instanceof CPacketPlayer.Rotation || packet instanceof CPacketPlayer.PositionRotation) {
+            this.serverRotation = new Rotation(packet.getYaw(0.0f), packet.getPitch(0.0f));
+        }
+    }
+
+    @Override
+    public void onWorldEvent(WorldEvent event) {
+        this.serverRotation = null;
+        this.target = null;
+    }
+
     public void pig() {
         if (this.target != null) {
-            ctx.player().rotationYaw = this.target.getYaw();
+            ctx.player().rotationYaw = this.target.rotation.getYaw();
+        }
+    }
+
+    public Optional<Rotation> getEffectiveRotation() {
+        if (Baritone.settings().freeLook.value || Baritone.settings().blockFreeLook.value) {
+            return Optional.ofNullable(this.serverRotation);
         }
+        // If neither of the freeLook settings are on, just defer to the player's actual rotations
+        return Optional.empty();
     }
 
     @Override
     public void onPlayerRotationMove(RotationMoveEvent event) {
         if (this.target != null) {
-
-            event.setYaw(this.target.getYaw());
-
-            // If we have antiCheatCompatibility on, we're going to use the target value later in onPlayerUpdate()
-            // Also the type has to be MOTION_UPDATE because that is called after JUMP
-            if (!Baritone.settings().antiCheatCompatibility.value && event.getType() == RotationMoveEvent.Type.MOTION_UPDATE && !this.force) {
-                this.target = null;
-            }
+            event.setYaw(this.target.rotation.getYaw());
         }
     }
 
     /**
      * Nudges the player's pitch to a regular level. (Between {@code -20} and {@code 10}, increments are by {@code 1})
      */
-    private void nudgeToLevel() {
-        if (ctx.player().rotationPitch < -20) {
-            ctx.player().rotationPitch++;
-        } else if (ctx.player().rotationPitch > 10) {
-            ctx.player().rotationPitch--;
+    private static float nudgeToLevel(float pitch) {
+        if (pitch < -20) {
+            return pitch + 1;
+        } else if (pitch > 10) {
+            return pitch - 1;
+        }
+        return pitch;
+    }
+
+    private static float calculateMouseMove(float current, float target) {
+        final float delta = target - current;
+        final int deltaPx = angleToMouse(delta);
+        return current + mouseToAngle(deltaPx);
+    }
+
+    private static int angleToMouse(float angleDelta) {
+        final float minAngleChange = mouseToAngle(1);
+        return Math.round(angleDelta / minAngleChange);
+    }
+
+    private static float mouseToAngle(int mouseDelta) {
+        final float f = Helper.mc.gameSettings.mouseSensitivity * 0.6f + 0.2f;
```

Comment:
On newer versions (checked 1.13 and 1.16) this setting is a `double`. Should it just be cast into a `float` (what I've done so far) or should all calculations be changed to use `double` instead?

Reply:
I believe the player's angles are still a float, so however the game handles the setting in the mouse callback code should be replicated.

Reply:
<details>
  <summary>Screenshots of the relevant game code from 1.19</summary>

![image](https://github.com/cabaletta/baritone/assets/24485393/5702ffb2-df9f-44bc-aa11-b142c43d3ec0)
![image](https://github.com/cabaletta/baritone/assets/24485393/e945c19a-8c51-4377-b336-f84e2cef1116)

</details>